lib/faraday_middleware/response/caching.rb

```diff
@@ -81,7 +88,11 @@ def cache_on_complete(env)
         # response.status is nil at this point, any checks need to be done inside on_complete block
         @app.call(env).on_complete do |response_env|
           if CACHEABLE_STATUS_CODES.include?(response_env.status)
-            cache.write(key, response_env.response)
+            if @options[:write_options]
+              cache.write(key, response_env.response, @options[:write_options])
+            else
+              cache.write(key, response_env.response)
+            end
```

Comment:
Lines 90-95 are exactly the same as lines 53-58, the only difference is the variable they use: `response_env.response` for the former, `response` for the latter (but they should be exactly the same).
Please refactor these lines into a method